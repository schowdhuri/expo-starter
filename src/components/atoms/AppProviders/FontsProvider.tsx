import {
  AlegreyaSansSC_100Thin,
  AlegreyaSansSC_100Thin_Italic,
  AlegreyaSansSC_300Light,
  AlegreyaSansSC_300Light_Italic,
  AlegreyaSansSC_400Regular,
  AlegreyaSansSC_400Regular_Italic,
  AlegreyaSansSC_500Medium,
  AlegreyaSansSC_500Medium_Italic,
  AlegreyaSansSC_700Bold,
  AlegreyaSansSC_700Bold_Italic,
  AlegreyaSansSC_800ExtraBold,
  AlegreyaSansSC_800ExtraBold_Italic,
  AlegreyaSansSC_900Black,
  AlegreyaSansSC_900Black_Italic,
} from "@expo-google-fonts/alegreya-sans-sc";
import {
  Roboto_100Thin,
  Roboto_100Thin_Italic,
  Roboto_300Light,
  Roboto_300Light_Italic,
  Roboto_400Regular,
  Roboto_400Regular_Italic,
  Roboto_500Medium,
  Roboto_500Medium_Italic,
  Roboto_700Bold,
  Roboto_700Bold_Italic,
  Roboto_900Black,
  Roboto_900Black_Italic,
} from "@expo-google-fonts/roboto";
import { useFonts } from "expo-font";
import type { FC } from "react";

export const FontsProvider: FC = ({ children }) => {
  const [fontsLoaded] = useFonts({
    AlegreyaSansSC_100Thin,
    AlegreyaSansSC_100Thin_Italic,
    AlegreyaSansSC_300Light,
    AlegreyaSansSC_300Light_Italic,
    AlegreyaSansSC_400Regular,
    AlegreyaSansSC_400Regular_Italic,
    AlegreyaSansSC_500Medium,
    AlegreyaSansSC_500Medium_Italic,
    AlegreyaSansSC_700Bold,
    AlegreyaSansSC_700Bold_Italic,
    AlegreyaSansSC_800ExtraBold,
    AlegreyaSansSC_800ExtraBold_Italic,
    AlegreyaSansSC_900Black,
    AlegreyaSansSC_900Black_Italic,
    Roboto_100Thin,
    Roboto_100Thin_Italic,
    Roboto_300Light,
    Roboto_300Light_Italic,
    Roboto_400Regular,
    Roboto_400Regular_Italic,
    Roboto_500Medium,
    Roboto_500Medium_Italic,
    Roboto_700Bold,
    Roboto_700Bold_Italic,
    Roboto_900Black,
    Roboto_900Black_Italic,
  });
  if (!fontsLoaded) {
    return <></>;
  }
  return <>{children}</>;
};
