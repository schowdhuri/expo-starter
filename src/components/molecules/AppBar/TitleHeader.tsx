import { Ionicons } from "@expo/vector-icons";
import { HStack, Icon, IconButton, Text, useTheme } from "native-base";

interface TitleWithBackProps {
  onGoBack?: () => void;
  title: string;
}

export function TitleHeader(props: TitleWithBackProps) {
  const { onGoBack, title } = props;
  const { colors } = useTheme();
  return (
    <HStack alignItems="center">
      <IconButton
        icon={
          <Icon
            size="lg"
            as={Ionicons}
            name="arrow-back"
            color={colors.gray[700]}
          />
        }
        onPress={onGoBack}
        testID="title-header"
      />
      <Text
        color={colors.primary[500]}
        fontSize={20}
        fontWeight="bold"
        lineHeight={20}
      >
        {title}
      </Text>
    </HStack>
  );
}
