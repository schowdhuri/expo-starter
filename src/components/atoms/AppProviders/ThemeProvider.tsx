import { NativeBaseProvider } from "native-base";
import type { FC } from "react";
import { theme } from "@constants/theme";
import { FontsProvider } from "./FontsProvider";

export const ThemeProvider: FC = ({ children }) => {
  return (
    <FontsProvider>
      <NativeBaseProvider initialWindowMetrics={INSET} theme={theme}>
        {children}
      </NativeBaseProvider>
    </FontsProvider>
  );
};

const INSET = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};
