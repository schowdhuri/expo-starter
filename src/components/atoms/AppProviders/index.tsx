import { NavigationContainer } from "@react-navigation/native";
import type { FC } from "react";
import { QueryClient, QueryClientProvider } from "react-query";
import { RecoilRoot } from "recoil";
import { ThemeProvider } from "./ThemeProvider";

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      staleTime: 30000,
      refetchOnWindowFocus: false,
    },
  },
});

export const AppProviders: FC = ({ children }) => {
  return (
    <QueryClientProvider client={queryClient}>
      <RecoilRoot>
        <ThemeProvider>
          <NavigationContainer>{children}</NavigationContainer>
        </ThemeProvider>
      </RecoilRoot>
    </QueryClientProvider>
  );
};
