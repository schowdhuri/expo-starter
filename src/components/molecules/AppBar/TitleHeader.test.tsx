import { render, waitFor, fireEvent } from "@testing-library/react-native";
import { AppProviders } from "@atoms/AppProviders";
import { TitleHeader } from "./TitleHeader";

describe("molecules::TitleHeader", () => {
  test("renders without exploding", async () => {
    const { toJSON, getByText, unmount } = render(
      <AppProviders>
        <TitleHeader onGoBack={jest.fn()} title="Title" />
      </AppProviders>
    );

    await waitFor(() => expect(getByText("Title")).toBeDefined());
    expect(toJSON()).toMatchSnapshot();
    unmount();
  });

  test("calls onPress", async () => {
    const spy = jest.fn();
    const { getByTestId, unmount } = render(
      <AppProviders>
        <TitleHeader onGoBack={spy} title="Title" />
      </AppProviders>
    );
    await waitFor(() => expect(getByTestId("title-header")).toBeDefined());
    expect(spy).toHaveBeenCalledTimes(0);
    fireEvent.press(getByTestId("title-header"));
    expect(spy).toHaveBeenCalledTimes(1);
    unmount();
  });
});
