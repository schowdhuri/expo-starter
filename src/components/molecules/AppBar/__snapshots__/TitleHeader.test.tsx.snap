// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules::TitleHeader renders without exploding 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      colorScheme="primary"
      dataSet={Object {}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 4,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
      testID="title-header"
    >
      <Text
        allowFontScaling={false}
        dataSet={Object {}}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#3f3f46",
              "fontSize": 24,
              "height": 24,
              "lineHeight": 24,
              "width": 24,
            },
            Object {
              "fontFamily": "ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ï„¤
      </Text>
    </View>
    <View
      dataSet={Object {}}
      style={Object {}}
    />
    <Text
      dataSet={Object {}}
      style={
        Object {
          "backgroundColor": undefined,
          "color": "#b69056",
          "fontFamily": "AlegreyaSansSC_700Bold",
          "fontSize": 20,
          "fontStyle": undefined,
          "fontWeight": undefined,
          "letterSpacing": 0,
          "lineHeight": 20,
          "textDecorationLine": undefined,
        }
      }
    >
      Title
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
